crates/conjure_rules/src/normalisers/weighted_sums.rs

```diff
@@ -43,6 +42,7 @@ fn collect_like_terms(expr: &Expr, _: &SymbolTable) -> ApplicationResult {
         match expr.clone() {
             Expr::Product(_, exprs2) => {
                 match exprs2.as_slice() {
+                    // todo (gs248) It would be nice to generate these destructures by macro, like `essence_expr!` but in reverse
```

Comment:
I have been looking into this, but it seems rust won't let you emit patterns with a macro. Haskell does, trivially. @niklasdewally also looked into it a bit and I think he found someting that was being worked on?